### ISSUE TITLE
Enum should implement IComparable and IFormattable

### ISSUE THREAD

**Issue by shadow-cs** [bridgedotnet/Bridge#2327]
**System.Enum** should implement **System.IComparable** and **System.IFormattable** interfaces, following code should compile.

### Steps To Reproduce

http://deck.net/179ce8e03ba33b8ca2d659a68c8ae3fa/top

```c#
public class App
{
    enum Foo
    {
        Bar, Baz
    }

    public static void Main()
    {
        Enum e = Foo.Bar;
        IComparable comparable = e;
        IFormattable formattable = e;

        Console.WriteLine(comparable.CompareTo(Foo.Baz));
        Console.WriteLine(comparable.CompareTo(Foo.Bar));

        Console.WriteLine("Bar", formattable.ToString("G", null));
        Console.WriteLine(e is IFormattable);
        Console.WriteLine(e is IComparable);
    }
}
```

### Expected Result

Successful compile and the following values in the console
- `-1`
- `0`
- `Bar`
- true
- true

See https://dotnetfiddle.net/IZ696a.

### Actual Result

```
Cannot implicitly convert type 'System.Enum' to 'System.IComparable'. An explicit conversion exists (are you missing a cast?)
```

**Comment by geoffreymcgill** [comment]
Hi @shadow-cs – Thanks for reporting this issue. We will investigate and make any changes that are required.

**Comment by geoffreymcgill** [comment]
@shadow-cs A pull request has been submitted with a fix for this issue. The pull request will go through a review and testing process, then once approved will be accepted into the **master** branch.

**Comment by geoffreymcgill** [comment]
This issue has been fixed and will be included in the 16.0 release.
